### PR TITLE
Calculate the database size from all files in the directory

### PR DESCRIPTION
**Pull request by yvonmanzi**
@badboy, I just fixed the issue. Thanks a lot for the guidance you provided on Bugzilla. Looking forward to some feedback.

### CODE REVIEW

File: glean-core/src/database/mod.rs [hunk 1]
```diff
@@ -104,6 +104,30 @@ fn file_size(path: &Path) -> Option<NonZeroU64> {
         .and_then(NonZeroU64::new)
 }
 
+/// Determine the size of all the files in the given file directory.
+///
+///  # Arguments
+///
+///  -`path` - The path to the directory
```
Comment (badboy) [inline glean-core/src/database/mod.rs:111]:
```suggestion
///  * `path` - The path to the directory
```

File: glean-core/src/database/mod.rs [hunk 2]
```diff
@@ -104,6 +104,30 @@ fn file_size(path: &Path) -> Option<NonZeroU64> {
...
+/// Returns the non-zero combined size of all files,
+/// or `None` on error or if the size is `0`.
+fn database_size(dir: &Path) -> Option<NonZeroU64> {
+    let mut total_size = 0;
+    if let Ok(entries) = fs::read_dir(dir) {
+        for entry in entries {
+            if let Ok(entry) = entry {
+                let path = entry.path();
+                // the getter is returning the u64 value from NonZeroU64
+                total_size += file_size(&path)?.get();
```
Comment (badboy) [inline glean-core/src/database/mod.rs:124]: We can savae some work here. You can read the [`metadata`](https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.metadata) of an entry directly, that one then has the [`len`](https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.len). You can then also remove the `file_size` function as its not used anymore.

If there's no metadata for whatever reason, we shouldn't bail out early here. If we can't read the file size of a single file, we should still read the remaining files.

File: glean-core/src/database/mod.rs [hunk 3]
```diff
@@ -104,6 +104,30 @@ fn file_size(path: &Path) -> Option<NonZeroU64> {
...
 
+/// Determine the size of all the files in the given file directory.
+///
+///  # Arguments
+///
+///  -`path` - The path to the directory
+///
+///  # Returns
+///
+/// Returns the non-zero combined size of all files,
```
Comment (badboy) [inline glean-core/src/database/mod.rs:115]:
```suggestion
/// Returns the non-zero combined size of all files in a directory,
```

File: glean-core/src/database/mod.rs [hunk 4]
```diff
@@ -104,6 +104,30 @@ fn file_size(path: &Path) -> Option<NonZeroU64> {
...
+/// or `None` on error or if the size is `0`.
+fn database_size(dir: &Path) -> Option<NonZeroU64> {
+    let mut total_size = 0;
+    if let Ok(entries) = fs::read_dir(dir) {
+        for entry in entries {
+            if let Ok(entry) = entry {
```
Comment (badboy) [inline glean-core/src/database/mod.rs:121]: An entry could be another directory.
We don't use nested directories here, so it _shouldn't happen. 

Let's still handle it by ignoring it explicitly. You can get an entries [`file_type`](https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.file_type) and check that `is_file` is true.

**Review by badboy — APPROVED** [review]
This is coming along nicely, but will need some adjustements.
See the inline comments.

Also please change the commit message: `Calculate the database size from all files in the directory` or something similar.

**Comment by yvonmanzi** [comment]
Awesome, makes sense. Thanks for the feedback! Lemme send another PR for you to see if it all looks good.

**Comment by badboy** [comment]
> Awesome, makes sense. Thanks for the feedback! Lemme send another PR for you to see if it all looks good.

You can update this PR (might require force-pushing to your branch)

**Comment by yvonmanzi** [comment]
Hi @badboy, I am not sure why this is happening, but I see that one of the checks is failing. I'd appreciate it if you helped me figure out what the issue is. Thanks.

**Comment by badboy** [comment]
> Hi @badboy, I am not sure why this is happening, but I see that one of the checks is failing. I'd appreciate it if you helped me figure out what the issue is. Thanks.

That seems unrelated and you can ignore it.
I somehow missed that you already pushed new code though, let me take a look.

**Review by badboy — APPROVED** [review]

**Pull request merged**